superset-frontend/src/explore/components/ExploreChartPanel.jsx

```diff
@@ -257,6 +258,16 @@ const ExploreChartPanel = props => {
     [chartPanelRef, renderChart],
   );
 
+  const [queryFormData, setQueryFormData] = useState(
+    props.chart.latestQueryFormData,
+  );
+
+  useEffect(() => {
+    if (!props.triggerRender) {
+      setQueryFormData(props.chart.latestQueryFormData);
+    }
+  }, [props.chart.latestQueryFormData]);
```

Comment:
Should we include `triggerRender` in deps array?

Reply:
No - that would cause it to trigger when we change controls whose `renderTrigger` prop is set to `true`.

Reply:
that would run `useEffect` function, but we'd exit at `if` check, right?

Reply:
Right you are - adding it!

Reply:
Actually, this causes a rerender when making changes to controls that don't have `renderTrigger` set to `true`:
https://user-images.githubusercontent.com/33317356/133249851-49e9ceb3-0865-4172-b915-0c2c2b0c5bdc.mp4

Reply:
Got it, thanks for explanation! In that case, can we add eslint suppress warning? I think it's time we started getting rid of those errors wherever possible 🙂

Reply:
I added a comment + supressed the warning 👍